matrix/src/lib.rs

```diff
@@ -82,3 +82,9 @@ pub trait MatrixRowSlices<T>: MatrixRows<T> {
 pub trait MatrixRowSlicesMut<T>: MatrixRowSlices<T> {
     fn row_slice_mut(&mut self, r: usize) -> &mut [T];
 }
+
+/// A `TransposeMatrix` which supports transpose logic for matrices
+pub trait MatrixTranspose<T>: MatrixRows<T> {
+    const BLOCK_SIZE: usize;
```

Comment:
Could we make this a private constant in `dense.rs`? It seems like more of an implementation detail to me

Reply:
Oh I see, the `match` couldn't be used if it's a global constant. I think we could also have a `block_size` local variable and const propagation should hopefully result in the same code.

Reply:
I refactored the code accordingly. Let me know your thoughts. Thanks for the suggestions !